api/v1alpha1/protectedapplicationview_types.go

```diff
@@ -0,0 +1,165 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ProtectedApplicationViewSpec defines the desired state of ProtectedApplicationView
+type ProtectedApplicationViewSpec struct {
+	// Type of application (ApplicationSet, Subscription, or Discovered)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=ApplicationSet;Subscription;Discovered
+	Type string `json:"type"`
+
+	// ApplicationRef references the source application resource
+	// +kubebuilder:validation:Required
+	ApplicationRef corev1.ObjectReference `json:"applicationRef"`
+
+	// Placements contains placement information for this application
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Placements []PlacementInfo `json:"placements,omitempty"`
+
+	// DRInfo contains disaster recovery configuration
+	// +kubebuilder:validation:Required
+	DRInfo DRInfo `json:"drInfo"`
+
+	// ApplicationSetInfo contains ApplicationSet-specific information
+	// +optional
+	ApplicationSetInfo *ApplicationSetInfo `json:"applicationSetInfo,omitempty"`
+
+	// SubscriptionInfo contains Subscription-specific information
+	// +optional
+	SubscriptionInfo *SubscriptionInfo `json:"subscriptionInfo,omitempty"`
```

Comment:
Considering "Subscription" based applications are going to be deprecated soon, we should rather have a generic type called "ApplicationInfo" which contains all application specific details.

Reply:
Changed to use applicationInfo and inside it we'll have subscriptioninfo